src/main/java/com/example/foode/Offer/Exception/OfferNotFoundAdvice.java

```diff
@@ -0,0 +1,19 @@
+package com.example.foode.Offer.Exception;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ControllerAdvice
+public class OfferNotFoundAdvice {
+
+    @ResponseBody
+    @ExceptionHandler(OfferNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    String offerNotFoundHandler(OfferNotFoundException ex) {
```

Comment:
Hmm is returning string for controller is good practise?

Reply:
I think that returning exception object is better but some reading here is needed

Reply:
+1